staging/src/k8s.io/kubectl/pkg/config/v1beta1/types.go

```diff
@@ -62,6 +65,32 @@ type Preference struct {
 	// "kubectl getn control-plane-1 --output=json" expands to "kubectl get node --output=json control-plane-1"
 	// +listType=atomic
 	Aliases []AliasOverride `json:"aliases"`
+
+	// credentialPluginPolicy specifies the policy governing which, if any, client-go
+	// credential plugins may be executed. It MUST be one of { "", "AllowAll", "DenyAll", "Allowlist" }.
+	// If the policy is "", then it falls back to "AllowAll" (this is required
+	// to maintain backward compatibility). If the policy is DenyAll, no
+	// credential plugins may run. If the policy is Allowlist, only those
+	// plugins meeting the criteria specified in the `credentialPluginAllowlist`
+	// field may run.
+	// +optional
+	CredentialPluginPolicy clientcmdapi.PolicyType `json:"credentialPluginPolicy,omitempty"`
+
+	// Allowlist is a slice of allowlist entries. If any of them is a match,
+	// then the executable in question may execute. That is, the result is the
+	// logical OR of all entries in the allowlist. This list MUST NOT be
+	// supplied if the policy is not "Allowlist".
+	//
+	// e.g.
+	// credentialPluginAllowlist:
+	// - name: cloud-provider-plugin
+	// - name: /usr/local/bin/my-plugin
+	// In the above example, the user allows the credential plugins
+	// `cloud-provider-plugin` (found somewhere in PATH), and the plugin found
+	// at the explicit path `/usr/local/bin/my-plugin`.
+	// +optional
+	// +listType=atomic
+	CredentialPluginAllowlist []clientcmdapi.AllowlistEntry `json:"credentialPluginAllowlist,omitempty"`
```

Comment:
This is currently referencing types defined in internal packages and counting on them having json tags that make them serializable... that's pretty confusing.

Keep the kuberc types all defined under kubectl/pkg/config/types.go and kubectl/pkg/config/v1beta1/types.go where they can follow normal behavior for json tags